### PR TITLE
Remove hardcoded supported packages list.

### PR DESCRIPTION
**Pull request by bdemers**
This can now be extended via a new service `ResourceFactoryConfig`

### CODE REVIEW

File: impl/src/main/java/com/okta/sdk/impl/ds/DefaultResourceFactory.java [hunk 1]
```diff
@@ -33,7 +38,7 @@
 
     private final InternalDataStore dataStore;
 
-    private static final List<String> SUPPORTED_PACKAGES = Arrays.asList("com.okta.sdk.", "com.okta.authn.sdk.");
```
Comment (bdemers) [inline impl/src/main/java/com/okta/sdk/impl/ds/DefaultResourceFactory.java]: @jmelberg-okta one of your comments on the previous review had me thinking.  There should not be any refs to `authn` in this repo if we can help it.

File: impl/src/main/java/com/okta/sdk/impl/ds/ResourceFactoryConfig.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+import java.util.Set;
+
+/**
+ * @since 1.1.0
+ */
+public interface ResourceFactoryConfig {
+
+    Set<String> getSupportedPackages();
+}
```
Comment (jmelberg-okta) [inline impl/src/main/java/com/okta/sdk/impl/ds/ResourceFactoryConfig.java]: You should add an `.editorconfig` file to detect missing or omit newlines. ;)

File: impl/src/main/java/com/okta/sdk/impl/ds/DefaultResourceFactory.java [hunk 3]
```diff
@@ -160,12 +165,22 @@ public DefaultResourceFactory(InternalDataStore dataStore) {
...
         }
 
         return args.toArray();
     }
+
+    private static Set<String> getSupportedPackages() {
+        Set<String> result = new HashSet<>();
+        stream(ServiceLoader.load(ResourceFactoryConfig.class).spliterator(), false)
+                    .map(config -> config.getSupportedPackages().stream()
+                               .map(it -> it +".")
```
Comment (jmelberg-okta) [inline impl/src/main/java/com/okta/sdk/impl/ds/DefaultResourceFactory.java]: `+".")` -> `+ ".")` (space)

File: impl/src/main/java/com/okta/sdk/impl/ds/ResourceFactoryConfig.java (hunk 2 above)
Comment (bdemers) [reply impl/src/main/java/com/okta/sdk/impl/ds/ResourceFactoryConfig.java]: It has one ;) though maybe it isn't configure right? it has: `insert_final_newline=false`

**Pull request merged**